src/lib_json/json_reader.cpp

```diff
@@ -1878,15 +1873,29 @@ class OurCharReader : public CharReader {
 
 public:
   OurCharReader(bool collectComments, OurFeatures const& features)
-      : collectComments_(collectComments), reader_(features) {}
-  bool parse(char const* beginDoc, char const* endDoc, Value* root,
-             String* errs) override {
-    bool ok = reader_.parse(beginDoc, endDoc, *root, collectComments_);
-    if (errs) {
-      *errs = reader_.getFormattedErrorMessages();
+      : collectComments_(collectComments), reader_(features) {
+        this->_impl=std::unique_ptr<Impl>(new Impl(collectComments, features));
+      }
+
+  class Impl : public CharReader::Impl {
+    bool const collectComments_;
+    OurReader reader_;
```

Comment:
On Line 1872, `OurCharReader` has an `OurReader` data member.
Isn't this `reader_` redundant with that one?

Reply:
not addressed, for example.

Reply:
I addressed this by removing the OurReader data member from OurCharReader